### PR TITLE
update `cody.chatResponse`'s `metadata` to include `lineCount` & `charCount`

### PR DESCRIPTION
**Pull request by akalia25**
Update `cody.chatResponse` event to include `lineCount` and `charCount` in metadata args, so that we can capture this data from more users. 

Since `metadata` can only accept numeric values, if `lineCount` or `charCount` is ever null their value will be -1. 
## Test plan
CI + locally
image

### CODE REVIEW

**Review by bobheadxi — APPROVED** [review]
Nice :-)

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts [hunk 1]
```diff
@@ -1035,21 +1035,23 @@ export class SimpleChatPanelProvider implements vscode.Disposable, ChatSession {
...
+            interactionID: requestID,
             metadata: {
+                ...generatedCode,
                 // Flag indicating this is a transcript event to go through ML data pipeline. Only for dotcom users
                 // See https://github.com/sourcegraph/sourcegraph/pull/59524
                 recordsPrivateMetadataTranscript: authStatus.isDotCom ? 1 : 0,
             },
             privateMetadata: {
-                ...hasCode,
                 requestID,
```
Comment (bobheadxi) [inline vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:1055]: Just out of curiosity: during our call we removed this in favour of `interactionID`, did you decide to put this back?

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts [hunk 2]
```diff
@@ -1035,21 +1035,23 @@ export class SimpleChatPanelProvider implements vscode.Disposable, ChatSession {
         const authStatus = this.authProvider.getAuthStatus()
 
         // Count code generated from response
-        const hasCode = countGeneratedCode(messageText.toString())
-        const responeEventName = hasCode?.charCount ? 'hasCode' : 'noCode'
+        const generatedCode = countGeneratedCode(messageText.toString())
+        const responseEventName = generatedCode.charCount > 0 ? 'hasCode' : 'noCode'
```
Comment (bobheadxi) [inline vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:1039]: Minor nit - maybe we should name this in a more V2-oriented manner, i.e. 

```suggestion
        const responseEventAction = generatedCode.charCount > 0 ? 'hasCode' : 'noCode'
```

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts (hunk 1 above)
Comment (akalia25) [reply vscode/src/chat/chat-view/SimpleChatPanelProvider.ts:1055]: Hey! Yeah, I decided to keep it until I get confirmation from @kelsey-brown, its okay to remove from `privateMetadata` :) ([thread](https://sourcegraph.slack.com/archives/C04MSD3DP5L/p1715710350117739?thread_ts=1715346003.349779&cid=C04MSD3DP5L))

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts (hunk 2 above)
Comment (akalia25) [reply vscode/src/chat/chat-view/SimpleChatPanelProvider.ts]: pushed this change 🚀

File: vscode/src/chat/chat-view/SimpleChatPanelProvider.ts (hunk 1 above)
Comment (akalia25) [reply vscode/src/chat/chat-view/SimpleChatPanelProvider.ts]: removed `requestId` from `privateMetadata` :D

**Pull request merged**